### ISSUE TITLE
MEP22 toolmanager does not set axes navigate_mode

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#18151]
### Bug report

**Bug summary**

Noted while looking at https://github.com/matplotlib/matplotlib/issues/18150: when using MEP22 toolmanager, `ax.get_navigate_mode()` always returns None even when a tool is active.

**Code for reproduction**

```python
ax = plt.gca()
# trigger the zoom tool
print(ax.get_navigate_mode())
```

**Actual outcome**

`None`

**Expected outcome**

`"zoom rect"`

**Matplotlib version**

  * Operating system: linux
  * Matplotlib version: master
  * Matplotlib backend (`print(matplotlib.get_backend())`): qt5agg
  * Python version: 38
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by Chinmay922** [comment]
I would like to look into this issue.

**Comment by tacaswell** [comment]
@Chinmay922 Great!  The code you are going to have to work with is in https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backend_tools.py please ask here or https://gitter.im/matplotlib/matplotlib if you have any questions.

**Comment by xordux** [comment]
Hello, 
I hope you are doing fine. I tried to reproduce it in my (windows) machine but I think it is not reproducing. As per the issue outcome should be None but I am getting "ZOOM" (not "zoom rect").

In the issue details I can see that "Linux" OS is mentioned. Is this a linux only issue? If yes, then which version of Linux I should try it on?

I am using Python 3.8 64 bit, if that matters.

**Comment by QuLogic** [comment]
Did you switch the toolbar rcParam to toolmanager first?

**Comment by xordux** [comment]
> Did you switch the toolbar rcParam to toolmanager first?

@QuLogic Thanks a lot. I didn't do that. Now the following code is reproducing the issue.

`import matplotlib.pyplot as plt
plt.ion()
plt.rcParams['toolbar'] = 'toolmanager'
ax = plt.gca()
# trigger the zoom tool in the generated plot
print(ax.get_navigate_mode())`

**Comment by tacaswell** [comment]
closed by #18493
